### PR TITLE
add hw hs1 week2 by Yash

### PR DESCRIPTION
**Pull request by ersarue**
typo in commit message, it should have been "add hw **js1** week2 by Yash"

### CODE REVIEW

**Comment by ersarue** [comment]
typo in commit message, it should have been 

> add hw **js1** week2 by Yash

File: hw-w2.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+console.log('the value of var4 is an ' typeof var4);
+
+// 8.5
+typeof var1 !== typeof var2; //returned: true
+typeof var3 == typeof var4; //returned: false
+
+// 8.6
+// with alert message
+if (typeof var1 !== typeof var4);
+alert('var1 and var4 are not the same type of variables!'); // returned pop-up message: var 1 and var4 are not the same type of variables!
```
Comment (nudge) [inline hw-w2.js:34]: nice use of `alert`!

File: hw-w2.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+
+// *** number 9 from Javascript1 week2 ***
+
+// 9: the new value of x would be 1, because the mathematical operator '%' provides with the remainder value when the assigned value is divided by the value at the right side of '%'.
+
+// 9.1 explanatory examples:
+a = 90; //returned 90
+b = a % 9; //returned 0
+c = a % 5; //returned 0
```
Comment (nudge) [inline hw-w2.js:48]: 👍

File: hw-w2.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+typeof var3 == typeof var4; //returned: false
+
+// 8.6
+// with alert message
+if (typeof var1 !== typeof var4);
+alert('var1 and var4 are not the same type of variables!'); // returned pop-up message: var 1 and var4 are not the same type of variables!
+
+// with console message
+if (typeof var1 !== typeof var4);
+console.log('var1 and var4 are not the same type of variables!'); // returned: var1 and var4 are not the same type of variables!
```
Comment (nudge) [inline hw-w2.js:38]: 👍

File: hw-w2.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+c = a % 5; //returned 0
+
+// *** number 10 from Javascript1 week2 ***
+
+// 10: yes, multiple types would be stored in an array.
+
+// 10.1
+myArray = [1, 2, 3, 'hahaha', null, [true, false]]; // returned: Array(6) [ 1, 2, 3, "hahaha", null, (2) […] ]
+
+// 10.2: comparing infinities depends on the type of value. in this case (6/0 === 10/0), it results in 'true' because === operator focuses on the types of the given values, not the values. here both of them is 'NaN'.
```
Comment (nudge) [inline hw-w2.js:57]: Close, both are in fact `Infinity` which is a pre-defined global variable, you can check this in the console!

File: hw-w2.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+console.log(myString.replace(/,/g, ' '));
+
+
+const str = 'dlroW olleH';
+function reverseString(str) {
+  const splitString = str.split('');
+  const reverseArray = splitString.reverse();
+  const joinArray = reverseArray.join('');
+  return joinArray;
+};
```
Comment (nudge) [inline hw-w2.js:87]: nice use of a function!

File: hw-w2.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+console.log(favoriteAnimals);
+
+favoriteAnimals.splice(1, 0, 'snake');
+console.log(favoriteAnimals); // returned: Array(5) [ "blowfish", "snake", "capricorn", "giraffe", "turtle" ]
+
+removedAnimal = favoriteAnimals.splice(3, 1)
+// returned: Array [ "giraffe" ]
+
+console.log(favoriteAnimals); //returned: Array(4) [ "blowfish", "snake", "capricorn", "turtle" ]
```
Comment (nudge) [inline hw-w2.js:114]: 👍

File: hw-w2.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+
+
+// *** Operators
+
+// if a == b and b == c, then a == c. this works because == operator focuses on the value, not the type. so after second statement b is equal of a and c values, meaning a is equivalent of c. but below
+
+0 == '0'; // true, coercing: JS converts each to a string when compared with == operator
+0 == [];  // true, because coercing into its numeric value so the length of [] is 0;
+"0" == [];  // false, because while "0" is falsy value, '[]' is truthy; making the result false (true + false = false)
```
Comment (nudge) [inline hw-w2.js:124]: this one is tricky, javascript coerces `[]` into an empty string `""` thus this expression becomes `"0" == ""` which is `false`.

File: hw-w2.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+// x == y false because references are different objects
+// x === y false because references are different objects even though their types are the same
+// z === y true because references are the same objects
+// z == x false because references are different objects, as z was assigned already to y (z === y)
```
Comment (nudge) [inline hw-w2.js:135]: 👍

File: hw-w2.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+
+const o1 = { foo: "bar" };
+const o2 = { foo: "bar" };
+const o3 = o2;
+
+// since the value of o3 is referenced to the value of o2, changing o2 will change o3.
+// changing o1 will not effect the value of o3 because no connection exists with o1
```
Comment (nudge) [inline hw-w2.js:143]: 👍

**Review by nudge — COMMENTED** [review]
Nice work!

**Pull request closed without merging**